### DISCUSSION TITLE
Compare and Diff arbitrary items from the same collection

### DISCUSSION THREAD

**Discussion by pemontto** [directus/directus#24049] [Feature Requests]
### Summary

Enhance Directus’s revision diff view functionality to allow users to compare different items within the same collection and arbitrary versions of those items. Additionally, provide an option to display the diff view in a full-size modal or dedicated page for side-by-side comparison.

![image](https://github.com/user-attachments/assets/f4f03517-49ab-4977-b2d6-7bd52b3f7601)

### Motivation

Currently, Directus’s diff view is limited to comparing revisions of a single version of a single item. Expanding this capability to include comparisons between different items within the same collection and between arbitrary versions of those items would:
- Support Use Cases: Enable users to identify differences between similar items, facilitating tasks such as content audits, quality control, and data consistency checks.
- Expected Outcome: Provide a more flexible and comprehensive diff tool that enhances data management and oversight within Directus.

### Detailed Design

1. **Enhanced Diff View Display**: Implement a full-size modal or dedicated page for the diff view, enabling side-by-side comparison of items or versions. This layout would improve readability and facilitate easier identification of changes.
1. **Comparing Different Items**: Introduce a “Compare Items” feature in the item list view, allowing users to select two items from the same collection for comparison. Upon selection, the system would display a side-by-side diff view highlighting differences in fields and content.
1. **Comparing Arbitrary Versions**: Within the enhanced diff view, provide dropdown menus for each side, enabling users to select specific versions of items to compare. This functionality would allow for:
    - **Same Item Comparison**: Users can select different versions of the same item on both sides to identify changes over time.
    - **Different Item Comparison:** Users can choose different items and their respective versions to compare, facilitating analysis of variations between items.

**Example Usage**:
- **Comparing Items**: In the collection view, users select two items and choose “Compare Items” from the action menu. The system displays a side-by-side diff view with dropdowns above each pane, allowing users to select specific versions of the chosen items for comparison.
- **Comparing Versions**: Within the diff view, users utilize the dropdowns to select different versions of the same item on both sides, enabling a detailed comparison of changes between those versions.

### Must Have:
- Full-size modal or dedicated page for side-by-side diff view.
- Ability to compare different items within the same collection.
- Dropdown menus in the diff view to select specific versions of items for comparison.

### Should Have:
- User-friendly interface for selecting items and versions to compare.
- Clear visual indicators highlighting differences.

### Could Have:
- Option to copy/update field from left version to right version
- Option to export diff results for documentation or reporting purposes.

### Won't Have:
- Comparison across different collections, as this may introduce complexity and is beyond the current scope.

### Drawbacks

Developing this feature would require significant changes to the current diff view functionality, including UI adjustments and backend logic to handle comparisons between different items and versions.

### Alternatives

- Users could export items and their versions to external diff tools for comparison. However, this approach is less efficient and disrupts the workflow within Directus.
- Organizations could develop custom modules or extensions to provide this functionality. This solution requires additional resources and may not integrate seamlessly with Directus’s core features.
- Maintaining the current functionality limits users’ ability to perform comprehensive comparisons, potentially affecting data management efficiency.

### Adoption Strategy

Implementing the proposed enhancements to Directus’s would be a new feature and non-breaking change.

### Unresolved Questions

Probably some crossover with https://github.com/directus/directus/discussions/20372 but this feature can be distinct from versions.

**Comment by burka** [comment]
I would love to see something like this :-)
